RealmSwift/Sync.swift

```diff
@@ -311,12 +311,43 @@ public typealias Provider = RLMIdentityProvider
         }
     }
 
+    /// The implementation of @dynamicMemberLookup that allows for dynamic remote function calls with a `callable` return.
+    public subscript(dynamicMember string: String) -> FunctionCallable {
+        FunctionCallable(name: string, user: user)
+    }
+}
+
+/// Structure providing an interface to been able to return a `callable` for a MongoDB Realm function.
+@dynamicCallable
+public struct FunctionCallable {
+    fileprivate let name: String
+    fileprivate let user: User
+
+    /// :nodoc:
+    public func dynamicallyCall(withArguments args: [Never]) {
+        /// Dummy implementation for @dynamicCallable
```

Comment:
I wouldn't actually use the word "Dummy" here 😅 . I would simply say `/// noop` and see if we can place it in the #else block beneath the #if block to better telegraph what this is doing.